qadence2_platforms/backends/analog/functions.py

```diff
@@ -67,6 +68,45 @@ def dyn_pulse(
     sequence.add(p, channel="global")
 
 
+def piecewise_pulse(
+    sequence: Sequence,
+    duration: Variable | VariableItem,
+    amplitude: Variable,
+    detuning: Variable,
+    phase: VariableItem | float,
+    **_: Any,
+) -> None:
+    """
+    Dynamic pulse to simulate a specific piecewise time-dependent hamiltonian for neutral-atom devices.
+
+    :param sequence: a `pulser.sequence.Sequence` instance
+    :param duration: duration of the pulse in nanoseconds
+    :param amplitude: amplitude of the pulse in rad/µs
+    :param detuning: detuning of the pulse in rad/s
+    :param phase: phase in rad
+    """
+    max_amp = sequence.device.channels["rydberg_global"].max_amp or DEFAULT_AMPLITUDE
+    max_abs_detuning = (
+        sequence.device.channels["rydberg_global"].max_abs_detuning or DEFAULT_DETUNING
+    )
+
+    dur_factor = 1000 * 2 * np.pi / max_amp  # type: ignore
+    amp_factor = max_amp  # type: ignore
+    det_factor = max_abs_detuning  # type: ignore
+
+    # Needed so a size = 1 variable is made iterable
```

Comment:
Open question: does the `PieceWiseDrive` need to support `size=1`, because in that case it is the same as `NativeDrive`, isn't it?

Reply:
Oh wait, a single duration can still be a ramp...